tools/liveimage-mount

```diff
@@ -142,155 +206,232 @@ def main():
-                dm_cow = 'live-rw'
-            else:
-                overlay = tempfile.NamedTemporaryFile(dir='/dev/shm')
-                print "\npreparing temporary overlay..."
-                call(['/bin/dd', 'if=/dev/null', 'of=%s' % overlay.name,
-                      'bs=1024', 'count=1', 'seek=%s' % (512 * 1024)])
-                call(['/sbin/losetup', overlayloop, overlay.name])
-                dm_cow = 'live-ro'
-            call(['/sbin/dmsetup', '--noudevrules', '--noudevsync',
-                 'create', dm_cow, '--table=0 %s snapshot %s %s p 8' %
-                 (imgsize, imgloop, overlayloop)])
-            call(['/bin/mount', os.path.join('/dev/mapper', dm_cow), destmnt])
-            home_path = os.path.join(liveosdir, 'home.img')
-            if os.path.exists(home_path):
-                homemnt = os.path.join(destmnt, 'home')
-                homeloop = rcall(losetup_args + [home_path]).rstrip()
-                call(['/bin/mount', homeloop, homemnt])
-            mntlive = os.path.join(destmnt, 'mnt', 'live')
-            if not os.path.exists(mntlive):
-                os.makedirs(mntlive)
-            call(['/bin/mount', '--bind', liveosmnt, mntlive])
-        elif img_type is 'iso':
-            imgloop = rcall(losetup_args + [ext3_img]).rstrip()
-            call(['/bin/mount', '-o', 'ro', imgloop, destmnt])
+            squashmnt = tempfile.mkdtemp(prefix='mp-squash-', dir='/run')
+            remove_mp += (squashmnt,)
+            mount(['-r', squash_img, squashmnt])
+            rootfs_img = os.path.join(squashmnt, 'LiveOS', 'ext3fs.img')
+            if not os.path.exists(rootfs_img):
+                rootfs_img = os.path.join(squashmnt, 'LiveOS', 'rootfs.img')
+            if not os.path.exists(rootfs_img):
+                print """
+                \r\tNo rootfs.img was found on %s\n\t  Exiting...\n""" % liveos
+                ecode = 1
+                return
+
+        imgloop = loop_setup(rootfs_img, '-r')
+        imgsize = rcall(['blockdev', '--getsz', imgloop])[0].rstrip()
+        dm_path = os.path.join('/dev', 'mapper', 'live-rw')
+        X = ''
+        if os.path.exists(dm_path):
+            fd, X = tempfile.mkstemp(prefix='live-rw-', dir='/dev/mapper')
+            os.close(fd)
+            os.unlink(X)
+            X = ''.join(('-', os.path.basename(X)[8:]))
+            dm_path += X
+        call(['dmsetup', 'create', '--readonly', 'live-base' + X,
+                              '--table=0 %s linear %s 0' % (imgsize, imgloop)])
+        for f in os.listdir(liveosdir):
+            if f.find('overlay-') == 0:
+                overlayloop = loop_setup(os.path.join(liveosdir, f))
+                break
+        persistent = 'P'
+        if roflag or 'overlayloop' not in locals():
+            persistent = 'N'
+            tmpoverlay = tempfile.NamedTemporaryFile(prefix='tmp-overlay-',
+                                                     dir='/run')
```

Comment:
I chose the temporary /run filesystem because that is where temporary overlays go when a LiveOS is booted.
To address the small memory system concern, a --tmpdir option has been added in a new version see pull request #15.